docs: sso group sync and supabase attribute mapping (#3803)
langchain-ai/docs: src/langsmith/user-management.mdx

      - `Default workspace role` and `Default workspaces` are editable. The updated settings will apply to new users only, not existing users.
      - (Coming soon) `SAML metadata URL` and `SAML metadata XML` are editable. This is usually only necessary when cryptographic keys are rotated/expired or the metadata URL has changed but the same IdP is still used.

### Supabase Attribute Mapping

<Note>

> Supabase Attribute Mapping is a cloud-only feature. Self-hosted deployments configure SAML/OIDC attributes directly with the IdP—see [Set up SSO with OAuth2.0 and OIDC](/langsmith/self-host-sso).

Reviewer: ```suggestion
Supabase Attribute Mapping is a [cloud-only](/langsmith/cloud) feature. [Self-hosted](/langsmith/self-hosted) deployments configure SAML/OIDC attributes directly with the IdP—see [Set up SSO with OAuth2.0 and OIDC](/langsmith/self-host-sso).
```

</Note>

LangSmith cloud uses [Supabase](/langsmith/cloud) as the SAML SSO backend. Supabase passes a small set of standard SAML attributes (such as `email` and `sub`) onto the user's JWT automatically. Any additional, non-standard SAML attribute your IdP emits (for example, `groups` for [SSO Groups Sync](#sso-groups-sync-alternative)) must be explicitly forwarded through Supabase before LangSmith can read it.

**Attribute flow (1:1):**

1. **IdP** — emits a SAML attribute with the configured name (e.g., `groups`).
2. **Supabase** — forwards the attribute onto the user's JWT only if the attribute name appears in the **Supabase Attribute Mapping** table on the SSO provider. Standard attributes are forwarded automatically; non-standard attributes are dropped unless explicitly listed.

> 3. **LangSmith** — reads the JWT claim by name (e.g., the value of [SSO Groups Sync](#sso-groups-sync-alternative)'s **Groups claim field**).

Reviewer: ```suggestion
1. **IdP**: emits a SAML attribute with the configured name (e.g., `groups`).
2. **Supabase**: forwards the attribute onto the user's JWT only if the attribute name appears in the **Supabase Attribute Mapping** table on the SSO provider. Standard attributes are forwarded automatically; non-standard attributes are dropped unless explicitly listed.
3. **LangSmith**: reads the JWT claim by name (e.g., the value of [SSO Groups Sync](#sso-groups-sync-alternative)'s **Groups claim field**).
```

#### Other identity providers

Other identity providers have not been tested but may function depending on their SCIM implementation.

### SSO Groups Sync (alternative)

<Note>

> SSO Groups Sync is available for organizations on the [Enterprise plan](https://www.langchain.com/pricing-langsmith) with SAML SSO (cloud) or OIDC (self-hosted) configured. [Contact sales](https://www.langchain.com/contact-sales) to learn more.

Reviewer: ```suggestion
SSO Groups Sync is available for organizations on the [Enterprise plan](/langsmith/pricing-plans) with SAML SSO (cloud) or OIDC (self-hosted) configured. [Contact sales](https://www.langchain.com/contact-sales) to learn more.
```

</Note>

> SSO Groups Sync is a lighter-weight alternative to [SCIM](#set-up-scim-for-your-organization) for organizations that can't or prefer not to configure SCIM group push. Instead of pushing groups from your IdP to LangSmith on a separate sync interval, LangSmith reads group memberships directly from a configurable claim in the SSO token at login time and applies org-level and workspace-level role assignments using the same [naming convention](#group-naming-convention) as SCIM.

Reviewer: ```suggestion
SSO Groups Sync is a simpler alternative to [SCIM](#set-up-scim-for-your-organization) for organizations that can't or prefer not to configure SCIM group push. Instead of pushing groups from your IdP to LangSmith on a separate sync interval, LangSmith reads group memberships directly from a configurable claim in the SSO token at login time and applies org-level and workspace-level role assignments using the same [naming convention](#group-naming-convention) as SCIM.
```

This is a well-established pattern used by GitLab (SAML Group Links), Grafana (Team Sync), HashiCorp Vault (`groups_claim`), and Atlassian (JIT group assignment).

#### When to use SSO Groups Sync vs. SCIM

> SSO Groups Sync and SCIM can technically coexist (each only manages identities tagged with its own provisioning method), but we recommend choosing **one mechanism per organization** — not both — to avoid confusing precedence behavior.

Reviewer: ```suggestion
SSO Groups Sync and SCIM can technically coexist (each only manages identities tagged with its own provisioning method), but we recommend choosing **one mechanism per organization**, not both, to avoid confusing precedence behavior.
```

| | SSO Groups Sync | SCIM |
| --- | --- | --- |
| **Sync trigger** | At each SSO login | Proactive push from IdP (~1 hour cadence) |

> | **IdP admin involvement** | Minimal — just include groups in the SSO token | Required — configure SCIM provisioning app |

Reviewer: ```suggestion
| **IdP admin involvement** | Minimal, just include groups in the SSO token | Required, configure SCIM provisioning app |
```

> | **Deprovisioning** | Lags until next login | Near-real-time via IdP push |

Reviewer: ```suggestion
| **Deprovisioning** | Lags until next login | Near real-time via IdP push |
```

| **Naming convention** | Reuses [SCIM convention](#group-naming-convention) | [SCIM convention](#group-naming-convention) |
| **Custom separator** | Reuses org-level [`scim_group_name_separator`](#configure-custom-separator) | [`scim_group_name_separator`](#configure-custom-separator) |

> Choose **SSO Groups Sync** when IdP admin involvement is minimal and reactive (login-time) sync is acceptable. Choose **SCIM** when proactive provisioning/deprovisioning and near-real-time group membership updates are required.

Reviewer: ```suggestion
Choose **SSO Groups Sync** when IdP admin involvement is minimal and reactive (login-time) sync is acceptable. Choose **SCIM** when proactive provisioning/deprovisioning and near real-time group membership updates are required.
```

   | --- | --- |
   | **Enable SSO Groups Sync** | Automatically assign workspace roles based on group memberships in the SSO token. |
   | **Groups claim field** (default `groups`) | The claim name in the SSO token that contains group memberships. |
   | **Sync workspace/role assignments** | Update workspace memberships and roles from group names on each SSO login. |
   | **Require matching group to sign in** | Block login if the SSO token contains no groups matching the naming convention. |

You can also configure these settings via the API by sending a `PATCH` to the SSO settings endpoint:

```bash
curl -X PATCH $LANGCHAIN_ENDPOINT/api/v1/orgs/current/sso-settings/$SSO_PROVIDER_ID \
  -H "X-Api-Key: $LANGCHAIN_API_KEY" \
  -H "Content-Type: application/json" \
  -d '{
    "sso_groups_enabled": true,
    "sso_groups_claim_field": "groups",
    "sso_groups_role_sync_enabled": true,
    "sso_groups_required": false
  }'
```

<Warning>
Disabling SSO Groups Sync does not remove existing access. Users provisioned by SSO groups will retain their current access until their next login.
</Warning>

#### Configure your IdP to emit a groups SAML attribute (cloud)

<Note>

> This section applies to **enterprise cloud** only. Self-hosted customers configure groups directly in their OIDC IdP — see [SSO Groups Sync on self-hosted](/langsmith/self-host-sso#sso-groups-sync).

Reviewer: ```suggestion
This section applies to **enterprise cloud** only. Self-hosted customers configure groups directly in their OIDC IdP, see [SSO Groups Sync on self-hosted](/langsmith/self-host-sso#sso-groups-sync).
```

</Note>

To make a user's group memberships visible to LangSmith at login, you need to do two things:

1. Configure your IdP's SAML application to emit a multi-valued group attribute.
2. Add a matching entry to [Supabase Attribute Mapping](#supabase-attribute-mapping) so the attribute flows through to the JWT (with **Array** checked).

**Requirements:**

- The IdP attribute name (e.g., `groups`) must match both the **Supabase Attribute Mapping** entry and the **Groups claim field** value (default `groups`).
- The attribute must be **multi-valued** (a list of strings), not a single delimited string. If your IdP only supports single-valued attributes, you'll need to emit one attribute statement per group.
- Each value must be a group name following the [SCIM naming convention](#group-naming-convention).

> - Only groups whose names match the convention are processed. Other groups (org-wide directory groups, app assignment groups, etc.) are skipped — emit them in the SAML attribute regardless if it's easier; LangSmith ignores irrelevant entries.

Reviewer: Could you take a look at that last sentence? I'm not sure if my suggestion fits with the meaning you want.
```suggestion
- Only groups whose names match the convention are processed. LangSmith ignores groups that don't match its naming convention, such as org-wide directory groups or app assignment groups. You don't need to filter these out on the IdP side—emit all groups and LangSmith will skip the irrelevant ones.
```

Author: > You don't need to filter these out on the IdP side—emit all groups and LangSmith will skip the irrelevant ones.

That is correct. Any groups like "Engineering" or "Hex Users" that are also sent over will be excluded. I'll accept as is.

| Workspace viewer in `Marketing` | `LS:Organization User:Marketing:Viewer` |
| Custom role `Annotators` in `Production` | `LS:Organization User:Production:Annotators` |

#### Behavior

- **Naming convention** — Group names follow the same format as SCIM (e.g., `LS:Organization Admins` for org admins, `LS:Organization User:Production:Editor` for workspace-scoped). See [Group naming convention](#group-naming-convention) for the full format. The separator is configured per-org via [`scim_group_name_separator`](#configure-custom-separator) and is shared with SCIM.
- **Malformed group names** — Group names that don't match the convention are skipped silently (logged) and don't block login for valid groups.
- **Login gate** — When **Require matching group to sign in** is enabled and the SSO token contains zero matching groups, login is blocked.
- **Precedence** — SCIM-sourced identities are never modified by SSO Groups Sync. Manually assigned and JIT-provisioned memberships are also untouched. SSO Groups Sync is fully authoritative for its own assignments and replaces them on each login based on the token's group membership.

> - **Org admin propagation** — If a user receives an org admin role from their groups, they are granted workspace admin in all workspaces (same as SCIM behavior).

Reviewer: ```suggestion
- **Naming convention**: Group names follow the same format as SCIM (e.g., `LS:Organization Admins` for org admins, `LS:Organization User:Production:Editor` for workspace-scoped). See [Group naming convention](#group-naming-convention) for the full format. The separator is configured per-org via [`scim_group_name_separator`](#configure-custom-separator) and is shared with SCIM.
- **Malformed group names**: Group names that don't match the convention are skipped silently (logged) and don't block login for valid groups.
- **Login gate**: When **Require matching group to sign in** is enabled and the SSO token contains zero matching groups, login is blocked.
- **Precedence**: SSO Groups Sync does not modify SCIM-sourced, manually assigned, or JIT-provisioned memberships. It is fully authoritative for its own assignments and replaces them on each login based on the token's group membership.
- **Org admin propagation**: If a user receives an org admin role from their groups, they are granted workspace admin in all workspaces (same as SCIM behavior).
```

#### Things to note

- **Deprovisioning lag** — Unlike SCIM (proactive push), SSO Groups Sync only updates on login. A user removed from a group in the IdP retains their existing workspace access until their next LangSmith login. The **Require matching group to sign in** gate mitigates this by blocking the user entirely on next login.
- **No retroactive sync** — Changing role mappings or enabling the feature does not update existing users until they log in again.

> - **Naming convention required** — Customers must name their IdP groups following the SCIM convention. If your IdP groups follow a different naming policy, SCIM with `description`-based mapping (see [Group attributes](#group-attributes)) may be a better fit.

Reviewer: ```suggestion
- **Deprovisioning lag**: Unlike SCIM (proactive push), SSO Groups Sync only updates on login. A user removed from a group in the IdP retains their existing workspace access until their next LangSmith login. The **Require matching group to sign in** gate mitigates this by blocking the user entirely on next login.
- **No retroactive sync**: Changing role mappings or enabling the feature does not update existing users until they log in again.
- **Naming convention required**: Customers must name their IdP groups following the SCIM convention. If your IdP groups follow a different naming policy, SCIM with `description`-based mapping (see [Group attributes](#group-attributes)) may be a better fit.
```
